packages/toolpad-app/src/toolpad/propertyControls/GridColumns.tsx

```diff
@@ -15,29 +15,52 @@ import {
   TextField,
   Tooltip,
 } from '@mui/material';
-import { GridColumns, GridColDef, GridAlignment } from '@mui/x-data-grid-pro';
 import * as React from 'react';
 import AddIcon from '@mui/icons-material/Add';
 import DeleteIcon from '@mui/icons-material/Delete';
 import ArrowBackIcon from '@mui/icons-material/ArrowBack';
 import RefreshIcon from '@mui/icons-material/Refresh';
-import { inferColumns } from '@mui/toolpad-components';
+import {
+  inferColumns,
+  NumberFormat,
+  NUMBER_FORMAT_PRESETS,
+  SerializableGridColumn,
+  SerializableGridColumns,
+} from '@mui/toolpad-components';
 import type { EditorProps } from '../../types';
 
 // TODO: this import suggests leaky abstraction
 import { usePageEditorState } from '../AppEditor/PageEditor/PageEditorProvider';
 import { generateUniqueString } from '../../utils/strings';
 
+type GridAlignment = SerializableGridColumn['align'];
+
 const COLUMN_TYPES: string[] = ['string', 'number', 'date', 'dateTime', 'boolean', 'link', 'image'];
 const ALIGNMENTS: GridAlignment[] = ['left', 'right', 'center'];
 
+function formatOptionValue(numberFormat: NumberFormat | undefined) {
```

Comment:
what about calling it `formatNumberOptionValue`, or something else that makes it clearer that this applies to numbers?